src/analysis/souffle/authorization_logic.dl

```diff
@@ -52,12 +52,33 @@ isPrincipal(speaker) :- saysDowngrades(speaker, _, _).
 //-----------------------------------------------------------------------------
 downgrades(path, tag) :- ownsTag(owner, tag), saysDowngrades(owner, path, tag).
 
+// This rule conservatively says that if _any_ principal claims that a path has
+// a tag, then it has the tag according to the taint analysis. This rule is
+// likely to be adjusted later. For confidentiality tags this should be OK to
+// start with.
+claimHasTag(path, tag) :- ownsTag(owner, tag), saysHasTag(_, path, tag).
+
+// This rule says that any principal can say that another principal owns a tag.
+// This rule is just a starting point to allow all aspects of the policy to
+// come from authorization logic -- it is just a simple way to generate
+// ownsTag from something with a speaker. This rule is likely to change. It
+// does not rule out the case that two different principals claim ownership of
+// a tag.
+ownsTag(speaker, tag) :- saysOwnsTag(speaker, owner, tag).
+
+//-----------------------------------------------------------------------------
+// Universe Populating Rules
+//-----------------------------------------------------------------------------
+
 // TODO(#126): Fix the universe populating rules, potentially by deleting
 //  them entirely.
-//
 // For now, we need to not populate isTag to handle cases where the tests
 // cause all principals to be associated with all tags.
 // isTag(tag) :- ownsTag(_, tag).
+
 isTag(tag) :- saysDowngrades(_, _, tag).
 
+isPrincipal(speaker) :- saysDowngrades(speaker, _, _).
+isPrincipal(owner) :- ownsTag(owner, _).
```

Comment:
We probably shouldn't introduce more universe populating rules that are based upon inferred relations, see #126.